preprocessing/regex_sentence_segmentation.py

```diff
@@ -0,0 +1,75 @@
+import re
+
+
+def sent_segmentation(text):
+    """Return a list of sentences from the text."""
+    
+    text = text.replace("\n", " ")
+    while "  " in text:
+        text = text.replace("  ", " ")
+
+    sentences = []
+    sent = ""
+
+    for i in range(len(text)-2):
+
+        # sentence can't start with end quotations marks
+        if len(sent) == 0 and text[i] == "\"“":
+            break
+
+        # check if not end of text
+        # check if char is interpunction
+        # check if previous char is not capital (no segmentation on abreviations)
+        # check if char after space is capital
+        # check beginning of next sentence
+        elif (i == len(text)-1 or i != len(text)-3 or i != len(text)-2 or i != 0) \
+            and (re.search(r"[.!?;:]", text[i]) \
+            and re.search(r"[^A-Z]", text[i-1]) \
+            and (re.search(r"[A-Z'“\"]", text[i+2])) \
+            or (re.search(r"[\"”]", text[i+1]) and re.search(r"[A-Z'“\"]", text[i+3]))):
+
+            # if next char end of quotation, append both
+            if re.search(r"[\"”]", text[i+1]):
+                sentences.append(sent + text[i] + text[i+1])
+                sent = ""
+            else:
+                sentences.append(sent + text[i])
+                sent = ""
+        else:
+            sent += text[i]
+
+    # append last sentence of the text + its interpunction
+    sentences.append(sent + text[len(text)-2:])
+
+    clean_sentences = []
+
+    for sent in sentences:
+
+        if sent[0] == " " or sent[0] == "\n":
```

Comment:
for this, we can use function `strip`
`sent = sent.strip() + "\n"`
It removes all kind of whitespaces from the beginning and end of string

Reply:
I tried that, but it didn't work the way I needed it to.